pthread_support.c

```diff
@@ -424,6 +425,16 @@ STATIC pthread_t GC_mark_threads[MAX_MARKERS];
         }
       }
 #   endif /* HPUX || GC_DGUX386_THREADS */
+
+    /* Apply special signal mask to GC marker threads, and don't drop user */
+    /* defined signals by GC marker threads.                               */
+    sigfillset(&set);
+    sigdelset(&set, GC_get_suspend_signal()    ); /* used by GC to stop the world    */
+    sigdelset(&set, GC_get_thr_restart_signal()); /* used by GC to restart the world */
+    if (pthread_sigmask(SIG_BLOCK, &set, &oldset) < 0) {
+      WARN("pthread_sigmask failed, errno = %" WARN_PRIdPTR "\n", errno);
```

Comment:
In case of pthread_sigmask failure, oldset is uninitialized so the second pthread_sigmask (below) call should not be performed. Also do we really want to start marker thread even if pthread_sigmask failed here? I suggest here to add: GC_markers_m1 = 0; (void)pthread_attr_destroy(&attr); return;